File: backend/app/models/verification.py
from pydantic import BaseModel, ConfigDict, Field


class ImageAnalysis(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    received_image: bool = Field(alias="receivedImage")
    image_type: str = Field(alias="imageType")
    summary: str


class SourceItem(BaseModel):
    title: str
    url: str
    relevance: str


class VerificationResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    trust_score: int = Field(alias="trustScore", ge=0, le=100)
    verdict: str
    risk_level: str = Field(alias="riskLevel")
    explanation: str
    content_type: str = Field(alias="contentType")
    detected_claims: list[str] = Field(alias="detectedClaims")
    image_analysis: ImageAnalysis = Field(alias="imageAnalysis")
    sources: list[SourceItem]


class HealthResponse(BaseModel):
    status: str
    service: str

File: extension/src/types/verification.ts
export interface ImageAnalysis {
  receivedImage: boolean
  imageType: string
  summary: string
}

export interface Source {
  title: string
  url: string
  relevance: string
}

export interface VerificationResponse {
  trustScore: number
  verdict: string
  riskLevel: string
  explanation: string
  contentType: string
  detectedClaims: string[]
  imageAnalysis: ImageAnalysis
  sources: Source[]
}

export interface VerifyPostInput {
  text?: string
  image?: File | null
  imageContext?: string
}
